src/sign-in-with-google/includes/class-sign-in-with-google-wpcli.php

```diff
@@ -0,0 +1,234 @@
+	 *     https://www.example.com?logmein // Send the user to authenticate with Google and log in
+	 *     https://www.example.com/my-custom-post?logmein // Log the user in and redirect to my-custom-post
+	 * ---
+	 *
+	 * [--show_on_login=<1|0>]
+	 * : Show the "Sign In With Google" button on the login form.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp siwg settings --client_id=XXXXXX.apps.googleusercontent.com
+	 *
+	 * @when after_wp_load
+	 */
+	public function settings( $args = array(), $assoc_args = array() ) {
+
+		// Sanitize everything.
+		$this->sanitize_args( $assoc_args );
+
+		// Update the Client ID.
+		if ( isset( $assoc_args['client_id'] ) ) {
+			$this->update_client_id( $assoc_args['client_id'] );
+		}
+
+		// Update the Client Secret.
+		if ( isset( $assoc_args['client_secret'] ) ) {
+			$this->update_client_secret( $assoc_args['client_secret'] );
+		}
+
+		// Update the default new user role.
+		if ( isset( $assoc_args['default_role'] ) ) {
+			$this->update_default_role( $assoc_args['default_role'] );
+		}
+
+		// Verify the list of domains and update the setting.
+		if ( isset( $assoc_args['domains'] ) ) {
+			$this->update_domain_restriction( $assoc_args['domains'] );
+		}
+
+		// Update the custom login parameter.
+		if ( isset( $assoc_args['custom_login_param'] ) ) {
+			$this->update_custom_login_param( $assoc_args['custom_login_param'] );
+		}
+
+		// Update show on login form.
+		if ( isset( $assoc_args['show_on_login'] ) ) {
+			$this->update_show_on_login_form( $assoc_args['show_on_login'] );
+		}
+
+		WP_CLI::success( 'Plugin settings updated' );
+
+	}
+
+	/**
+	 * Handles updating siwg_google_client_id.
+	 *
+	 * @param string $client_id The ID to use with Google's Oauth.
+	 */
+	private function update_client_id( $client_id = '' ) {
+		if ( '' == $client_id ) {
+			WP_CLI::error( 'Please enter a valid Client ID' );
+		}
+
+		$result = update_option( 'siwg_google_client_id', $client_id );
+
+		if ( ! $result ) {
+			WP_CLI::warning( 'Skipping Client ID - Setting already matches' );
+		}
+	}
+
+	/**
+	 * Handles updating siwg_google_client_secret.
+	 *
+	 * @param string $client_secret The secret to use with Google's Oauth.
+	 */
+	private function update_client_secret( $client_secret = '' ) {
+		if ( '' == $client_secret ) {
+			WP_CLI::error( 'Please enter a valid Client Secret' );
+		}
+
+		$result = update_option( 'siwg_google_client_secret', $client_secret );
+
+		if ( ! $result ) {
+			WP_CLI::warning( 'Skipping Client Secret - Setting already matches' );
+		}
+	}
+
+	/**
+	 * Handles updating siwg_google_user_default_role.
+	 *
+	 * @param string $role The role applied for new users.
+	 */
+	private function update_default_role( $role = 'subscriber' ) {
+		if ( '' == $role ) {
+			WP_CLI::error( 'Please enter a valid user role' );
+		}
+
+		if ( 'subscriber' != $role ) {
+
+			// All role names are lowercase.
+			$role = strtolower( $role );
+
+			$result = update_option( 'siwg_google_user_default_role', $role );
```

Comment:
You should make sure that the role exists before saving it as an option.

Reply:
Nice thinking!